src/compression/snappy.rs

```diff
@@ -1,18 +1,37 @@
 use std::io::{self, Read};
 
 use byteorder::{BigEndian, ByteOrder};
-use snappy;
+use snap;
 
 use error::{Result, Error};
 
 pub fn compress(src: &[u8]) -> Result<Vec<u8>> {
-    Ok(snappy::compress(src))
+    let mut buf = vec![0; snap::max_compress_len(src.len())];
+
+    snap::Encoder::new()
+        .compress(src, &mut buf)
+        .map(|len| {
+                 buf.truncate(len);
+
+                 buf
+             })
+        .map_err(|err| Error::InvalidSnappy(err))
 }
 
 fn uncompress_to(src: &[u8], dst: &mut Vec<u8>) -> Result<()> {
-    snappy::uncompress_to(src, dst)
-        .map(|_| ())
-        .map_err(|_| Error::InvalidInputSnappy)
+    snap::decompress_len(src)
+        .and_then(|min_len| {
+            let off = dst.len();
```

Comment:
@BurntSushi many thanks for the quick response! i agree, it first has to prove it is worth while.